sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2030,7 +2030,10 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession {
       def verifyCallCount(df: DataFrame, expectedResult: Row, expectedCount: Int): Unit = {
         countAcc.setValue(0)
         QueryTest.checkAnswer(
-          df, Seq(expectedResult), checkToRDD = false /* avoid duplicate exec */)
+          df, Seq(expectedResult), checkToRDD = false /* avoid duplicate exec */) match {
```

Comment:
Why not adding a new method in `QueryTest` object to let it call `fail` instead? The pattern is simply duplicated across changes, and even `QueryTest.checkAnswer` (protected method), which warrants enough reason to have another method.

e.g. `checkAnswer(df: DataFrame, expectedAnswer: java.util.List[Row]): String`

Reply:
Otherwise, maybe simply call checkAnswer (without QueryTest) - other places in this suite already called like this. Same applies for other suites as well.